requirements/0004-volunteer--host-identity-validated.md

```diff
@@ -0,0 +1,175 @@
+
+
+# Summary
+
+This is an RFC to expose public form for host self-registration, with identity validation functionality.
+
+# Motivation
+
+To be able to replace Salam Lab's Google Form, but also add a required level of protection against abuse, by requiring strong identity validation of *Hosts*.
+
+# Detailed Design
+
+## Use of IETF Keywords
+
+This document employs a subset of the Internet Engineering Task Force keywords found in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119). These words are MUST, SHALL, SHOULD, MAY and their counterparts MUST NOT, SHOULD NOT, MAY NOT. They are capitalized throughout the document to draw attention to their special status as keywords used to indicate requirements levels.
+
+Readers are directed to interpret them as requirements at levels consistent with their term of art definitions in the IETF [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Business Architecture
+
+### Capabilities
+
+Provide functionality for *Hosts* to self-register, and for the *Volunteer* to see if the *Host* of *Accommodation Unit* they are about to send the *Guests* too is a legitimate person.
+
+### Assumptions
+
+1. Most identity verification strategies take some time, and are asynchronous by nature
+
+### Constraints
+
+1. Applicable GDPR laws: https://gdpr-info.eu/
+
+### Dependencies
+
+The system needs to be able to verify the identity of the *Host*. The project SHALL not aim at providing solution to this problem as there are vendors in the market offering such solutions.
+
+Considered options are:
+   - [Blue Media](https://bluemedia.pl/) - 1pln money transfer loop, openbanking-based verification, "Selfie"-biometrics verification, Liveness check biometric verification.
+   - [Authologic](https://authologic.com/pl) - banking KYC, 1gr money transfer loop, "Selfie" biometrics, email verification, polish cell number verification, foreign cell number verification, digital id w/ PIN, digital ID w/o PIN.
+   - [Autenti](https://autenti.com/pl/) - the offer seemed to include only e-signature verification, and didn't suit project needs.
+   - [Stripe](https://stripe.com/docs/identity/verification-sessions) - the world-famous [Ukraine Take Shelter](https://www.ukrainetakeshelter.com/) uses it. Not entirely sure what is their pricing strategy. Might be free, but it's highly unreadable. This option popped up after decision was made.
+
+The system SHALL perform verification via [Authologic](https://authologic.com).
+
+## Risks
+
+### Business Risks
+
+1. No methods of identity validation are 100% foolproof. We are using the state-of-the-art solutions, but an evil actor determined enough, can bypass _in person_ validation at the bank, let alone a remote one like ours.
+
+### Technical Risks
+
+1. Authologic Terms of Use allow up to 1 minute of downtime without notice.
+
+### Impacts
+
+None.
+
+## Data Architecture
+
+`HostVerificationSession` table SHALL be introduced, with following fields:
```

Comment:
Do we want to store (effectively "cache") some other upstream fields from the conversation here? Like url, userKey, status, the verified first & last name & national ID/bank account number?

Reply:
url? probably not. url's are derived from conversation_id... userKey, my thought was that `guid` of `HostVerificationSession` will be used as `userId` - bc `userKey` is essentialy an id that we give to authologic so we can tie conversation back to our db. 

As for bank data... I was hesitant to store it because of RODO, etc, but maybe I shouldn't as we already store bunch of data.

Reply:
Ah, so your idea was to retrieve national/bank ID data only on demand if there's a need to pass them to authorities? I'll ask Authologic if the conversation data stays forever and what sort of durability estimates they have.